dvc/repo/init.py

```diff
@@ -63,9 +73,19 @@ def init(root_dir=os.curdir, no_scm=False, force=False):
     Raises:
         KeyError: Raises an exception.
     """
+
+    if no_scm and subrepo:
+        raise InitError("Cannot initialize repo with `--no-scm` and `--sub`")
+
     root_dir = os.path.realpath(root_dir)
+
+    if not subrepo:
+        scm_root_dir = root_dir
+    else:
+        scm_root_dir = find_scm_dir(root_dir)
+
     dvc_dir = os.path.join(root_dir, Repo.DVC_DIR)
-    scm = SCM(root_dir)
+    scm = SCM(scm_root_dir)
```

Comment:
See, you are spreading the context here again, I think it could be moved nicely to scm/, which will also avoid code duplication that you have here and in `Repo.__init__`